src/clarity.ts

```diff
@@ -34,17 +35,25 @@ export function start(override: Config = {}): void {
   }
 }
 
-export function pause(): void {
+function restart(): void {
+  start();
+}
+
+export function suspend(): void {
   end();
-  bind(document, "mousemove", resume);
-  bind(document, "touchstart", resume);
-  bind(window, "resize", resume);
-  bind(window, "scroll", resume);
-  bind(window, "pageshow", resume);
+  bind(document, "mousemove", restart);
+  bind(document, "touchstart", restart);
+  bind(window, "resize", restart);
```

Comment:
Will multiple calls to restart be handled correctly (potentially a race condition if there are multiple events queued up.

Reply:
This is not new logic so if it was handled before, it will continue to be the same deal. However going through the code we were unnecessarily re-initializing modules twice on multiple start calls. I've added the check to prevent that from happening. Thanks for pointing it out.

---
In reply to: [368665036](https://github.com/microsoft/clarity-js/pull/166#discussion_r368665036) [](ancestors = 368665036)